chapter_5_practic/task6.cpp

```diff
@@ -0,0 +1,59 @@
+// Start with the program from Exercise 11 in Chapter 4, �Structures, � which adds two
+// struct time values.Keep the same functionality, but modify the program so that it uses
+// two functions.The first, time_to_secs(), takes as its only argument a structure of type
+// time, and returns the equivalent in seconds(type long).The second function,
+// secs_to_time(), takes as its only argument a time in seconds(type long), and returns a
+// structure of type time.
+
+#include <iostream>
+
+struct user_time
+{
+	int hours = 0;
+	int minutes = 0;
+	int seconds = 0;
+};
+
+user_time time1;
+user_time time2;
+
+const int sec_in_hour = 3600;
+const int sec_in_min = 60;
+char ch = ' ';
+long totalsec = 0;
+
+long time_to_secs(user_time);
+user_time secs_to_time(long);
+
+int main()
+{
+	std::cout << "Enter time in format hh/mm/ss: " << std::endl;
+	std::cin >> time1.hours >> ch >> time1.minutes >> ch >> time1.seconds;
+	totalsec = time_to_secs(time1);
+	time2 = secs_to_time(totalsec);
+
+	std::cout << time2.hours << ":" << time2.minutes << ":" << time2.seconds;
+
+	return EXIT_SUCCESS;
+}
+
+long time_to_secs(user_time t)
+{
+	totalsec = t.hours * sec_in_hour + t.minutes * sec_in_min + t.seconds;
+	return totalsec;
+}
+
+user_time secs_to_time(long total)
+{
+	time2.hours = total / sec_in_hour;
+	time2.minutes = 0;
```

Comment:
logic is not optimal
The number of seconds is the reminder of total / 3600
The number of minutes is the reminder (total - sec) / 60
The number of hours is the total / 60

Reply:
Fixed, but I got 4 actions.

for example 50000 seconds:

hours: 50000 / 3600 = 13 hours;
seconds: total % 3600 = 3200 reminder of seconds
minutes: 3200 / 60 = 53 minutes;  
seconds = 3200 % 60 = 20 seconds; 

https://github.com/Dmitriy-Romanchuk/master/pull/5/commits/799c73b05317bbba47a2b293b1aab739bbd44172